.github/workflows/general.yml

```diff
@@ -8,11 +8,9 @@ on:
   pull_request:
     branches: ["main"]
 
-# When triggered from the merge queue, cancel any existing workflow runs for the same PR branch
-# Otherwise, use the unique run id for the concurrency group, to prevent anything from getting cancelled
 concurrency:
-  group: ${{ github.event_name == 'merge_group' && format('{0}-{1}', github.workflow, github.ref) || github.run_id }}
-  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
This will still cause workflows to get cancelled on `main` if we have more than 2 commits merged at once (a concurrency group only allows one running workflow and one pending working - any new ones will get cancelled).

I think we need to set:
```
concurrency:
  group: ${{ github.ref == 'refs/heads/main' && github.run_id || format('{0}-{1}', github.workflow, github.ref) }}
```

This will give each workflow run on `main` a unique concurrency group (so nothing ever gets cancelled), and group other runs by source branch.